### PR TITLE
[1LP][RFR]Adding new tests of vm reconfig add sockets and memory on running Vm

### PR DESCRIPTION
**Pull request by niyazRedhat**
Purpose or Intent
=================

{{ pytest: cfme/tests/infrastructure/test_vm_reconfigure.py -k test_vm_reconfig_add_remove_hw_hot --use-template-cache --long-running -sqvvv }}

### CODE REVIEW

File: requirements/frozen.py2.txt [hunk 1]
```diff
@@ -323,7 +323,7 @@ Werkzeug==0.14.1
 widgetastic.core==0.32.1
 widgetastic.patternfly==0.0.41
 widgetsnbextension==3.4.2
-wrapanapi==3.0.43
+git+git://github.com/digitronik/wrapanapi.git@hot-add
```
Comment (mshriver) [inline requirements/frozen.py2.txt]: :+1:

Comment (mshriver) [reply requirements/frozen.py2.txt]: https://pypi.org/project/wrapanapi/

3.0.44 includes the hot-swap properties needed for this test, please update all relevant requirements files.

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/tests/infrastructure/test_vm_reconfigure.py [hunk 2]
```diff
@@ -215,7 +225,11 @@ def test_vm_reconfig_add_remove_hw_hot_vmware(change_type):
             4.Change number of CPU sockets and amount of memory, save and submit
             5. Check the count in VM details page
     """
-    pass
+    orig_config = small_vm.configuration.copy()
+    new_config = prepare_new_config(orig_config, change_type)
+
+    # Apply new config
+    reconfigure_vm(small_vm, new_config)
```
Comment (sshveta) [inline cfme/tests/infrastructure/test_vm_reconfigure.py]: Any assertions that we can do to check if VM is reconfigured ?

File: cfme/tests/infrastructure/test_vm_reconfigure.py [hunk 3]
```diff
@@ -196,16 +204,18 @@ def test_vm_reconfig_add_remove_hw_hot(provider, small_vm, ensure_vm_running, ch
     reconfigure_vm(small_vm, orig_config)
 
 
-@pytest.mark.manual
 @pytest.mark.tier(1)
 @pytest.mark.parametrize('change_type', ['sockets', 'memory'])
-def test_vm_reconfig_add_remove_hw_hot_vmware(change_type):
-    """
+@pytest.mark.provider([VMwareProvider], override=True)
```
Comment (sshveta) [inline cfme/tests/infrastructure/test_vm_reconfigure.py]: Does this test apply for RHEV too ?

Comment (niyazRedhat) [reply cfme/tests/infrastructure/test_vm_reconfigure.py]: No, we already have test for RHEV 
https://github.com/ManageIQ/integration_tests/blob/8a8f3f954b9e76c32ae443dea724d1a34a1de7a3/cfme/tests/infrastructure/test_vm_reconfigure.py#L180

File: cfme/tests/infrastructure/test_vm_reconfigure.py (hunk 2 above)
Comment (niyazRedhat) [reply cfme/tests/infrastructure/test_vm_reconfigure.py]: added assertions

File: cfme/tests/infrastructure/test_vm_reconfigure.py (hunk 3 above)
Comment (sshveta) [reply cfme/tests/infrastructure/test_vm_reconfigure.py]: Test steps seems same .. can we reuse RHEV test for Vmware too and mark it to run for both providers?

Comment (niyazRedhat) [reply cfme/tests/infrastructure/test_vm_reconfigure.py]: @sshveta In vmware we enabling hot_plugin and in RHV we don't need that also in wmware we can't revert the changes just like we doing for RHV
https://github.com/ManageIQ/integration_tests/blob/8a8f3f954b9e76c32ae443dea724d1a34a1de7a3/cfme/tests/infrastructure/test_vm_reconfigure.py#L195

Comment (izapolsk) [reply cfme/tests/infrastructure/test_vm_reconfigure.py]: @niyazRedhat, either way this sounds like a code duplication.  
you can add if provider.one_of and run or don't run provider specific code.

**Comment by sshveta** [comment]
Please resolve PRT failure .

**Comment by sshveta** [comment]
> Please resolve PRT failure .



> Please resolve PRT failure .

test_vm_reconfig_add_remove_hw_hot[rhevm-4.3-sockets] FAILED

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/tests/infrastructure/test_vm_reconfigure.py [hunk 4]
```diff
@@ -47,7 +47,7 @@ def reconfigure_vm(vm, config):
 
 
 @pytest.fixture(scope='function')
-def small_vm(appliance, provider, small_template):
+def small_vm(appliance, provider, full_template):
```
Comment (izapolsk) [inline cfme/tests/infrastructure/test_vm_reconfigure.py]: shouldn't this fixture be removed along with full_template ?

File: cfme/tests/infrastructure/test_vm_reconfigure.py [hunk 5]
```diff
@@ -222,30 +232,11 @@ def test_vm_reconfig_add_remove_hw_hot(provider, small_vm, ensure_vm_running, ch
     # Apply new config
```
Comment (izapolsk) [inline cfme/tests/infrastructure/test_vm_reconfigure.py]: REQUIRED : please add assert before reconfiguration in order to see that values are really different.

**Pull request merged**